landscape/sysinfo/processes.py

```diff
@@ -14,10 +14,20 @@ def register(self, sysinfo):
     def run(self):
         num_processes = 0
         num_zombies = 0
-        info = ProcessInformation(proc_dir=self._proc_dir)
-        for process_info in info.get_all_process_info():
+        for pid in os.listdir(self._proc_dir):
+            if not pid.isdigit():
+                continue
+            status_path = os.path.join(self._proc_dir, pid, "status")
+            try:
+                fd = os.open(status_path, os.O_RDONLY)
+                try:
+                    data = os.read(fd, 2048)
+                finally:
+                    os.close(fd)
```

Comment:
The whole block should use "modern" context manager. Also, the use of human-readable `/status` instead of the machine parse-able `/stat` seems like a poor choice. I assume this was just copied and trimmed from l/lib/process.py, but the proper way would probably be like:

```
try:
    with open(status_path, "rb") as fd:
        data = f.read()
except IOError:
    continue
num_processes += 1
if b"Z" == data.split(b" ", 3)[2]:
    num_zombies += 1
```

One or two zombies tests in `landscape/sysinfo/tests/test_processes.py` would need an extra `create_data(…, stat_data="0 0 Z 0 0 0 0")` argument to make them pass.